src/ipc/ipc-common.c

```diff
@@ -138,9 +138,6 @@ struct ipc_comp_dev *ipc_get_ppl_comp(struct ipc *ipc, uint32_t pipeline_id, int
 		if (icd->type != COMP_TYPE_COMPONENT)
 			continue;
 
-		if (!cpu_is_me(icd->core))
```

Comment:
Good find. I guess we dont have such topologies in use today ?

Reply:
> Good find. I guess we dont have such topologies in use today ?

@lgirdwood I'm not aware of any, no, and in fact there are other problems with such configurations. After this such configurations begin to work, but some of them still remain racy. So we have to decide - best and simplest would be to say, that no cross-core connections are allowed (all connected pipelines and their components must run on the same core) and remove any signs of support for such cases, or we should consider and fix them properly.

Reply:
Cross core is allowed especially on Zephyr, bugs will be fixed as they arise.